fund_store/scripts/data_updates/FLS-907-update-la-names.py

```diff
@@ -0,0 +1,525 @@
+                )
+                application.project_name = CODE_TO_NAME_MAP[application.project_name]
+            else:
+                print(f"Application.project_name not found: {application.project_name}")
+
+            for form in application.forms:
+                if form.name not in [
+                    "local-plans-local-authority-details-signed-off",
+                    "local-plans-local-authority-details",
+                ]:
+                    continue
+
+                new_json = form.json
+                for category in new_json:
+                    for field in category["fields"]:
+                        if field["key"] not in ["RoLhhf", "OlCBjB"]:
+                            continue
+
+                        if field["answer"] in CODE_TO_NAME_MAP:
+                            print(
+                                f"Form.answer: {field['answer']} ",
+                                f"replacing with: {CODE_TO_NAME_MAP[field['answer']]}",
+                            )
+                            field["answer"] = CODE_TO_NAME_MAP[field["answer"]]
+                        else:
+                            print(f"From answer not found: {field['key']}, {field['answer']}")
+
+                # update the json
+                form.json = new_json
+
+        assessment_records = db.session.scalars(
+            select(AssessmentRecord).where(AssessmentRecord.fund_id.in_([round.fund_id]))
+        ).all()
+        print(f"\n\n{fund.short_name} AssessmentRecord count: {len(assessment_records)}\n")
+
+        for assessment_record in assessment_records:
+            project_name = assessment_record.project_name
+            json_project_name = assessment_record.jsonb_blob["project_name"]
+
+            print(f"\nAssessmentRecord: {assessment_record.application_id}, {project_name}")
+            if project_name in CODE_TO_NAME_MAP:
+                print(
+                    f"AssessmentRecord.project_name: {project_name} ",
+                    f"replacing with: {CODE_TO_NAME_MAP[project_name]}",
+                )
+                project_name = CODE_TO_NAME_MAP[project_name]
+            else:
+                print(f"AssessmentRecord.project_name not found: {project_name}")
+
+            new_json_2 = copy.deepcopy(assessment_record.jsonb_blob)
+
+            print(
+                f"AssessmentRecord.json_project_name: {assessment_record.application_id},",
+                f"{json_project_name}",
+            )
+            if json_project_name in CODE_TO_NAME_MAP:
+                print(
+                    f"AssessmentRecord.json_project_name: {json_project_name} ",
+                    f"replacing with: {CODE_TO_NAME_MAP[json_project_name]}",
+                )
+                new_json_2["project_name"] = CODE_TO_NAME_MAP[json_project_name]
+            else:
+                print(f"AssessmentRecord.json_project_name not found: {json_project_name}")
```

Comment:
Are the project names not being found in either of the database row or the JSON blob valid states for assessment? I would think both are necessarily there and this could be replaced by just updating the project name value on the model + the JSON 

Not a problem to be verbose in a script like this but I'd think these if conditions are almost a reason to halt the script rather than be managed like this

Reply:
No need to halt it.

In a world were we re-run the script, the already updated ONS Codes would be mentioned, but not updated.
Also, while the component is a drop-down, null or any string can be potentially stored. In this case, I am happy for it to be echo'd, but would not want to stop the sprint.